### PR TITLE
Fix USB class/subclass IDs on darwin

### PR DESCRIPTION
**Pull request by barn-stripe**
Mirrors linux's class/subclass output now.

### CODE REVIEW

**Comment by fmanco** [comment]
Just removed the issue from the title, otherwise it gets confusing on the final commit.

Fixes #4348.

File: osquery/tables/system/darwin/usb_devices.cpp [hunk 1]
```diff
@@ -26,6 +26,14 @@ std::string decodeUSBBCD(uint16_t bcd) {
   return std::to_string(major) + "." + std::to_string(minor);
 }
 
+// Decoding the class/usbclass returns.
+std::string decodeUSBBCDToInt(uint16_t bcd) {
```
Comment (fmanco) [inline osquery/tables/system/darwin/usb_devices.cpp]: Could you please detail what you're trying to achieve here?

Comment (barn-stripe) [reply osquery/tables/system/darwin/usb_devices.cpp]: Yeah, I spent ages trying to get the correct numbers out of what getNumIOKitProperty() returned, and how it's encoded going by https://beyondlogic.org/usbnutshell/usb5.shtml#DeviceDescriptors, and eventually made that function to get them in the right 0-255 range like the linux table returned for the same devices.

Comment (obelisk) [reply osquery/tables/system/darwin/usb_devices.cpp]:
```
  r["class"] = std::to_string(getNumIOKitProperty(details, "bDeviceClass") & 0xFF);
  r["subclass"] = std::to_string(getNumIOKitProperty(details, "bDeviceSubClass") & 0xFF);
```

Would this achieve the same thing for you?

**Comment by barn-stripe** [comment]
```osquery> select vendor, vendor_id, version, model,model_id,serial,class,subclass from usb_devices;
+--------------+-----------+---------+---------------------------------+----------+--------------+-------+----------+
| vendor       | vendor_id | version | model                           | model_id | serial       | class | subclass |
+--------------+-----------+---------+---------------------------------+----------+--------------+-------+----------+
| Apple Inc.   | 05ac      | 0.0     | Root Hub Simulation Simulation  | 8006     | 0            | 9     | 255      |
| Apple Inc.   | 05ac      | 0.0     | Root Hub Simulation Simulation  | 8007     | 0            | 9     | 255      |
|              | 0e0f      | 1.0     | VMware Virtual USB Hub          | 0002     | 0            | 9     | 0        |
| VMware       | 0e0f      | 1.2     | VMware Virtual USB Mouse        | 0003     | 0            | 0     | 0        |
|              | 0e0f      | 1.0     | VMware Virtual USB Hub          | 0002     | 0            | 9     | 0        |
| VMware       | 0e0f      | 1.0     | Virtual Bluetooth Adapter       | 0008     | 000650268328 | 224   | 1        |
| VMware, Inc. | 0e0f      | 1.0     | VMware Virtual USB Video Device | 000b     | 0            | 239   | 2        |
| VMware       | 05ac      | 1.0     | VMware Virtual USB Keyboard     | 020b     | 0            | 0     | 0        |
+--------------+-----------+---------+---------------------------------+----------+--------------+-------+----------+``` with @fmanco's better suggestion. Making an awe inspiring two line patch (:

**Comment by fmanco** [comment]
Thank you @barn-stripe

**Pull request merged**